examples/medplum-photon-integration/src/bots/sync-patient-from-photon.ts

```diff
@@ -0,0 +1,446 @@
+            name {
+              title
+              first
+              last
+            }
+            email
+            phone
+            address {
+              street1
+              street2
+              postalCode
+              country
+              state
+              city
+            }
+            organizations {
+              id
+              name
+            }
+          }
+          state
+          treatment {
+            id
+            codes {
+              rxcui
+            }
+          }
+          dispenseAsWritten
+          dispenseQuantity
+          refillsAllowed
+          refillsRemaining
+          fillsAllowed
+          fillsRemaining
+          daysSupply
+          instructions
+          notes
+          effectiveDate
+          expirationDate
+          writtenAt
+          fills {
+            id
+            treatment {
+              id
+              codes {
+                rxcui
+              }
+            }
+            state
+            requestedAt
+            filledAt
+          }
+        }
+        address {
+          street1
+          street2
+          postalCode
+          country
+          state
+          city
+        }
+      }
+    }
+  `;
+
+  const body = JSON.stringify({ query });
+
+  const result = await photonGraphqlFetch(body, photonAuthToken);
+  const photonPatients = result.data.patients as PhotonPatient[];
+  const batch: Bundle = {
+    resourceType: 'Bundle',
+    type: 'batch',
+    entry: [],
+  };
+
+  for (const photonPatient of photonPatients) {
+    // Make sure the patient does not already exist
+    let patient: Patient | undefined = await getExistingPatient(photonPatient, medplum);
+    if (!patient) {
+      // Create the patient resource
+      patient = createPatientResource(photonPatient);
+    }
+    const patientUrl = 'urn:uuid:' + randomUUID();
+    const patientReference: Reference<Patient> = { reference: patientUrl, display: getDisplayString(patient) };
+
+    // Create any allergies the patient has
+    const allergies = createAllergies(patientReference, photonPatient.allergies);
+
+    // Create any prescriptions
+    const prescriptions = await createPrescriptions(patientReference, medplum, photonPatient.prescriptions);
+
+    // Add the patient resource to a bundle
+    const patientEntry: BundleEntry = {
+      fullUrl: patientUrl,
+      request: { method: 'PUT', url: `Patient?identifier=${NEUTRON_HEALTH_PATIENTS}|${photonPatient.id}` },
+      resource: patient,
+    };
+    batch.entry?.push(patientEntry);
+
+    // If there are allergies, create entries and add them to the bundle
+    if (allergies) {
+      const allergyEntries: BundleEntry[] = allergies.map((allergy) => {
+        return {
+          fullUrl: 'urn:uuid:' + randomUUID(),
+          request: {
+            method: 'PUT',
+            url: `AllergyIntolerance?_source=${NEUTRON_HEALTH}&code=${allergy.code}&patient=${allergy.patient.reference}`,
```

Comment:
I got some guidance that we shouldn't be using `urn:uuid` references in the `url` during an upsert. 

Instead, we should be using chained search like this: 
`patient.identifier=${NEUTRON_HEALTH_PATIENTS}|${photonPatient.id}`